InnerEye/ML/configs/classification/CovidHierarchicalModel.py

```diff
@@ -0,0 +1,241 @@
+import codecs
+import logging
+import pickle
+from pathlib import Path
+
+from typing import Any, Callable
+
+import PIL
+import pandas as pd
+import param
+import torch
+from PIL import Image
+from pytorch_lightning import LightningModule
+from torchvision.transforms import Compose
+
+from InnerEye.Common.common_util import ModelProcessing, get_best_epoch_results_path
+from InnerEye.ML.SSL.datamodules_and_datasets.transforms_utils import create_chest_xray_transform
+from InnerEye.ML.SSL.lightning_containers.ssl_container import EncoderName
+
+from InnerEye.ML.SSL.lightning_modules.ssl_classifier_module import SSLClassifier
+from InnerEye.ML.SSL.utils import create_ssl_encoder, create_ssl_image_classifier, load_ssl_augmentation_config
+from InnerEye.ML.common import ModelExecutionMode
+
+from InnerEye.ML.configs.ssl.CXR_SSL_configs import path_linear_head_augmentation_cxr
+from InnerEye.ML.deep_learning_config import LRSchedulerType, MultiprocessingStartMethod, \
+    OptimizerType
+
+from InnerEye.ML.model_config_base import ModelTransformsPerExecutionMode
+from InnerEye.ML.model_testing import MODEL_OUTPUT_CSV
+
+from InnerEye.ML.models.architectures.classification.image_encoder_with_mlp import ImagingFeatureType
+from InnerEye.ML.reports.notebook_report import generate_notebook, get_ipynb_report_name, str_or_empty
+
+from InnerEye.ML.scalar_config import ScalarLoss, ScalarModelBase
+from InnerEye.ML.utils.augmentation import ScalarItemAugmentation
+from InnerEye.ML.utils.run_recovery import RunRecovery
+from InnerEye.ML.utils.split_dataset import DatasetSplits
+
+from InnerEye.ML.configs.ssl.CovidContainers import COVID_DATASET_ID
+from InnerEye.Common import fixed_paths as fixed_paths_innereye
+
+
+class CovidHierarchicalModel(ScalarModelBase):
+    """
+    Model to train a CovidDataset model from scratch or finetune from SSL-pretrained model.
+
+    For AML you need to provide the run_id of your SSL training job as a command line argument
+    --pretraining_run_recovery_id=id_of_your_ssl_model, this will download the checkpoints of the run to your
+    machine and load the corresponding pretrained model.
+
+    To recover from a particular checkpoint from your SSL run e.g. "recovery_epoch=499.ckpt" please use hte
+    --name_of_checkpoint argument.
+    """
+    use_pretrained_model = param.Boolean(default=False, doc="TBD")
+    freeze_encoder = param.Boolean(default=False, doc="Whether to freeze the pretrained encoder or not.")
+    name_of_checkpoint = param.String(default=None, doc="Filename of checkpoint to use for recovery")
+    test_set_ids = param.String(default=None,
```

Comment:
better to indicate in the attribute name that this refers to a CSV file, e.g. `test_set_csv`, `test_ids_csv`, or `test_set_ids_csv`